### PR TITLE
add network upgrades to chain ctx

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Network upgrade configs are dynamic for non-production (mainnet, fuji) + non-local chains. Chains should be aware that any changed network upgrade and configure themselves accordingly. 

Related Coreth PR: https://github.com/ava-labs/coreth/pull/630/files

## How this works

This adds the specified (or default) network upgrade configs to chain context.

## How this was tested

Locally (with coreth + subnet-evm)

### CODE REVIEW

File: proto/vm/vm.proto [hunk 1]
```diff
@@ -116,6 +116,8 @@ message InitializeRequest {
   // the messenger, keystore, shared memory, blockchain alias,
   // subnet alias, and appSender services
   string server_addr = 14;
+  // network_upgrades_bytes is the json encoded network upgrades
+  bytes network_upgrades_bytes = 15;
```
Comment (darioush) [inline proto/vm/vm.proto:120]: does this need an RPC version bump?

**Review by darioush — APPROVED** [review]

File: proto/vm/vm.proto (hunk 1 above)
Comment (ceyonur) [reply proto/vm/vm.proto:120]: yes

Comment (StephenButtolph) [inline proto/vm/vm.proto:120]: It seems like an anti-pattern to encode a json blob into the proto. Can we create an upgrade message that specifies the actual upgrade times?

Comment (ceyonur) [reply proto/vm/vm.proto]: added

File: vms/rpcchainvm/vm_server.go [hunk 2]
```diff
@@ -843,3 +851,76 @@ func (vm *VMServer) StateSummaryAccept(
 		Err:  errorToErrEnum[err],
 	}, errorToRPCError(err)
 }
+
+func convertNetworkUpgrades(pbUpgrades *vmpb.NetworkUpgrades) (upgrade.Config, error) {
```
Comment (StephenButtolph) [inline vms/rpcchainvm/vm_server.go:855]: `pbUpgrades` may be `nil` here - Can we error rather than panic in that case?

**Review by StephenButtolph — COMMENTED** [review]
1 nit then lgtm

File: vms/rpcchainvm/vm_server.go (hunk 2 above)
Comment (ceyonur) [reply vms/rpcchainvm/vm_server.go:856]: added 👍

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**